src/Compilers/CSharp/Portable/Binder/PatternSwitchBinder.cs

```diff
@@ -245,30 +248,529 @@ internal override void BindPatternSwitchLabelForInference(CasePatternSwitchLabel
                         {
                             diagnostics.Add(ErrorCode.ERR_DuplicateCaseLabel, node.Location, label.Name);
                             hasErrors = true;
+                            return new BoundPatternSwitchLabel(node, label, pattern, null, hasErrors);
+                        }
+                        else
+                        {
+                            // Note that this is semantically last! The caller will place it in the decision dag
+                            // in the final position.
+                            return defaultLabel = new BoundPatternSwitchLabel(node, label, pattern, null, hasErrors);
                         }
-
-                        // We always treat the default label as reachable, even if the switch is complete.
-                        const bool isReachable = true;
-
-                        // Note that this is semantically last! The caller will place it in the decision tree
-                        // in the final position.
-                        defaultLabel = new BoundPatternSwitchLabel(node, label, pattern, null, isReachable, hasErrors);
-                        return defaultLabel;
                     }
 
                 case SyntaxKind.CasePatternSwitchLabel:
                     {
                         var matchLabelSyntax = (CasePatternSwitchLabelSyntax)node;
-                        var pattern = sectionBinder.BindPattern(
+                        BoundPattern pattern = sectionBinder.BindPattern(
                             matchLabelSyntax.Pattern, SwitchGoverningType, node.HasErrors, diagnostics);
                         return new BoundPatternSwitchLabel(node, label, pattern,
                             matchLabelSyntax.WhenClause != null ? sectionBinder.BindBooleanExpression(matchLabelSyntax.WhenClause.Condition, diagnostics) : null,
-                            true, node.HasErrors);
+                            node.HasErrors);
                     }
 
                 default:
                     throw ExceptionUtilities.UnexpectedValue(node);
             }
         }
+
+        #region from old SwitchBinder.cs
```

Comment:
>region from old SwitchBinder.cs [](start = 9, length = 31)

Perhaps leave this code in that file, so it's clear what actual changes were from `SwitchBinder`.

Reply:
Good idea